heroku_connect/models/fields.py

```diff
@@ -0,0 +1,177 @@
+"""
+Salesforce fields for Django's ORM.
+
+You can find a list of all Salesforce fields mapped to PostgreSQL here.
+See Heroku Connect's `mapped data types`_.
+
+.. _`mapped data types`:
+    https://devcenter.heroku.com/articles/heroku-connect-database-tables#mapped-data-types
+
+"""
+from django.db import models
+
+__all__ = (
+    'HerokuConnectFieldMixin', 'AnyType', 'ID', 'Checkbox', 'Currency',
+    'Date', 'DateTime', 'Email', 'EncryptedString', 'Number', 'Percent',
+    'Phone', 'Picklist', 'Text', 'TextArea', 'Time', 'URL',
+)
+
+
+class HerokuConnectFieldMixin:
+    """Base mixin for Heroku Connect fields."""
+
+    #: (str): Field's Salesforce API name.
+    sf_field_name = None
+    #: (bool): Whether or not a field is an ``externalId``.
+    upsert = False
```

Comment:
I get what this is doing after reading the [relevant doc][], but I think it could use a beefy doc string to explain what this does: It marks the single, optional field that will be used by Salesforce as an external ID, in order to avoid race conditions, and which must therefore be unique and unchanging.

The naming (`upsert`/ `upsert_field`) does not make things less confusing; it might be better to name the attribute something along the lines of `is_external_id` or even `is_the_external_id`.

[relevant doc]: https://devcenter.heroku.com/articles/heroku-connect#unique-identifier

Reply:
`upsert` is a cryptic name indeed. Is there any specific reason for calling it that way? If not, I agree with @tilboerner that `is_the_external_id` or sth along the lines would be better.

Reply:
Good point, this does need improvement. It also doesn't follow the `sf_` prefix as we have for all other attributes. I guess `upsert` is a bad name. It is only used in Heroku Connect, not in salesforce. Ideally this library doesn't require much Heroku Connect knowledge but only Salesforce and Django (the whole point of this library really).

Reply:
I would love a suggestion.

Reply:
We can also move it to the model. There is only one upsert field anyways.

Reply:
Hm... that is the difference really. You can have n externalIDs but only one can be an upsert field....